light/beacon/api/syncer.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2022 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package api
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/light/beacon"
+	"github.com/ethereum/go-ethereum/log"
+	lru "github.com/hashicorp/golang-lru"
+)
+
+const (
+	headPollFrequency = time.Millisecond * 200
+	headPollCount     = 50
+	maxRequest        = 8
+)
+
+// CommitteeSyncer syncs committee updates and signed heads from BeaconLightApi to SyncCommitteeTracker
+type CommitteeSyncer struct {
+	api *BeaconLightApi
+
+	genesisData         beacon.GenesisData
+	checkpointPeriod    uint64
+	checkpointCommittee []byte
+	sct                 *beacon.SyncCommitteeTracker
+
+	updateCache, committeeCache *lru.Cache
```

Comment:
Let's use the new generic cache instead!

Reply:
done